blackboxopt/visualizations/visualizer.py

```diff
@@ -195,6 +195,85 @@ def multi_objective_visualization(
     return fig
 
 
+def parallel_coordinate_plot_parameters(
+    evaluations: List[Evaluation], color_by: Optional[Objective] = None
+):
+    if not evaluations:
+        raise NoSuccessfulEvaluationsError
+
+    # Prepare dataframe for visualization
+    df = evaluations_to_df(evaluations)
+
+    # Drop unused columns and indices
+    df = df[["configuration", "objectives"]]
+    parameter_cols = df["configuration"].columns.to_list()
+    objective_cols = df["objectives"].columns.to_list()
+    color_by_col = color_by.name if color_by else objective_cols[0]
```

Comment:
Since there is no necessarily unified order in the evaluations' objective values dictionaries I would consider taking the "first" as the default to be too arbitrary and would rather always require an explicit objective to color by. But then again not having to specify it is nice for the single objective cases, so let's leave it as is. And it also feels weird to only require it for the multi objective case (i.e. raise value error if multiple objective values and `color_by=None`, or what do you think?

Reply:
Honestly, after fiddling around a bit, I'm not convinced anymore that showing multiple objectives in one chart is a good idea, and I start preferring the initial solution more and more:

1. **Chart type:** To me it feels like a parallel coordinates plot is not suitable to plot an additionally quality (type: parameter vs. objective). The main reason is, that it relies on the coloring as key: The color is the thing that connects all coordinates. We can only color one coordinate, so there will never be a connection between a non-colored-objective and the parameters, (except for the ones directly next to it, where the lines give an indication). Such a plot would mainly allow comparing the colored objective to each individual other objective. For that purpose, we have the more suitable pareto-plot. Comparing the the non-colored objectives to the parameters will always be hard/cumbersome in such a chart.
2. **Developer UX:** If we allow plotting only one explicitly required objective, everything is totally transparent. As soon as we want to support multiple objectives, we'd have to make assumptions or handle different cases.
3. **Visual:** The colored objective is visually separated from the parameters. As soon as we add more objectives, we kind of start putting apples and oranges on the same visual level, putting an indicator to the label-name is just a mediocre workaround.

Another options, that would leave the decision to user, 
Require `objective:Union[Objective, List[Objective]], color_by: Optional[Objective]=None` and only plot the objectives provided. In case a list of objectives is provided but no `color_by` is set, the first objective in the list will be used for coloring. 

For the multi objective case, my dream chart would be multi-row, with one paracord chart per row for each objective, but with the ability to crossfilter (if you select some evaluations in one of the rows, they are also selected in the others)). Unfortunately, this is time consuming to implement. I could try to approximate this by returning a multi-row chart without crossfilter, but that might need some time. It would certainly have the benefit, that the `color_by`/`objective` arg would become obsolete.

What do you think about those thoughts?

Reply:
I feel you, so here is a proposal that should be flexible enough to not artificially restrict use cases with multiple objectives while also allowing for a smooth UX in the simple single objective case while not causing us devs much pain if complicated case handling. But before I go there I just want to highlight that there is value in a version of this plot without colors because one still gets the option to subselect and see a vague notion of "density". That being said, here is the proposal:

```python
def parallel_coordinate_plot_parameters(
    evaluations: List[evaluations],
    columns: Optional[List[str]] = None,
    color_by: Optional[str] = None
)
```

Where columns can be None for all available or a specified subset of the columns (config keys, settings keys, objective keys). And we only color the plot if `color_by` is set, and it can be any column of the data.

This allows for the most prominent use cases I currently see...

1) simple single objective case
```python
parallel_coordinate_plot_parameters(evals, color_by: "loss")
```

2) comparing only objectives for moo case without color
```python
parallel_coordinate_plot_parameters(evals, columns=["loss", "params", "flops"])
```

3) having few objectives in one and optionally use one to color
```python
parallel_coordinate_plot_parameters(evals) # or
parallel_coordinate_plot_parameters(evals, color_by="chosen objective")
```

Reply:
**RE: column**
Yes, documentation - also, I am hoping that it helps for people to see the unrestricted state, because there they see the actual vertical lines i.e. columns for this to additionally drive home the point what columns are conceptually.
Also, please include settings keys if they aren't already (most prominently fidelity), because that's something to interactively filter by for sure.

**RE: cumbersome use case**
Could you elaborate on the insights you derive from a visualization of multi objective run results where parameters and only one objective are shown? Usually, I find one does multi objective optimization because one is interested in a trade off - i.e. looking at the combination of the objectives, as opposed to dropping the information about all but one objective. Can also be a direct conversation separate from this PR, since it's not blocking it.

Reply:
Thanks for the trust and the discussion - I hope it'll be rewarded with a good experience down the line :relaxed:

Reply:
Could you please review my latest commit which implements your proposal? Especially the docstring... :innocent:

Reply:
>  Also, please include settings keys if they aren't already (most prominently fidelity), because that's something to interactively filter by for sure.

Ah, good idea! I'll do that tomorrow :-D

Reply:
Just to add another data point for the more flexible solution, which I like very much.
In the AIMO context, we saw what our colleagues would look at for MOO results, and they would predominantly look at a parallel coordinate plot with only the objectives. They didn't want to see the actual configuration (it was too high dimensional and the parameters where hard to interpret), but wanted a single plot to see and compare different trade-offs. This would now also be possible with the latest change :)

Reply:
Implemented.